cloud-init/postfix-setup.tpl.yml

```
@@ -0,0 +1,6 @@
+---
+runcmd:
+  # Modify postfix's PRIMARY_DOMAIN in pca-gophish-composition docker-compose
+  # file to the email-sending domain for this assessment
+  - [sed, -i, "s|PRIMARY_DOMAIN=.*|PRIMARY_DOMAIN=${email_sending_domain}|g",
```

This regex will apply to lines that look like `.*PRIMARY_DOMAIN=.*$`.  I suspect that you want it to apply only to lines that look like `^PRIMARY_DOMAIN=[^\w]*$`.  I suggest changing the regex to this to be extra safe:
```suggestion
  - [sed, -i, "s|^PRIMARY_DOMAIN=[^\w]*$|PRIMARY_DOMAIN=${email_sending_domain}|g",
```

---

Based on the `docker-compose.yml` this is likely editing, at least part of your edit would not work: https://github.com/cisagov/pca-gophish-composition/blob/47c69034a5df627852444932ed960660aeadce2b/docker-compose.yml#L67

---

Ah, your eyes are as sharp as ever @mcdonnnj!  How about this then:
```suggestion
  - [sed, -i, "s|^\([- \w]*\)PRIMARY_DOMAIN=[^\w]*$|\1PRIMARY_DOMAIN=${email_sending_domain}|g",
```

---

Thanks @mcdonnnj, you beat me to the punch on the `docker-compose.yml` stuff. 😄  

@jsf9k, your most-recent suggestion is almost perfect, but I had to escape the backslashes to appease the Python YAML linter used by cloud-init as well as add a `\` linebreak to satisfy our pre-commit `yamllint` hook.  

I verified that it worked as expected by deploying it to my test environment, then made the change in https://github.com/cisagov/cool-assessment-terraform/pull/112/commits/4d2c4c29757ea4fabfd570a78323e2f40c78408b.

Thanks for the improvement!